subprojects/core/src/main/java/org/gradle/api/internal/StartParameterInternal.java

```diff
@@ -36,17 +36,17 @@
     private boolean configurationCacheQuiet;
 
     @Override
-    public StartParameter newInstance() {
-        return prepareNewInstance(new StartParameterInternal());
+    public StartParameterInternal newInstance() {
+        return (StartParameterInternal) prepareNewInstance(new StartParameterInternal());
```

Comment:
I guess `prepareNewInstance()` could be abstract in `StartParamter` and have an override to return `StartParameterInternal` here.

Also, `prepareNewInstance()` must include copying the internal values like `watchFileSystem`. Maybe that field should move to `StartParameter`? Or we need an override of the method here that also copies internal values? This needs to be taken into account for configuration caching as well. This class hierarchy is a complete mess in any case. :/

Also, maybe rename `prepareNewInstace()` to `makeCopy()` or something, as that's what it does?

Reply:
`prepareNewInstance` can't be abstract in `StartParameter`, since `StartParameter` isn't an abstract class. Note that `prepareNewInstance` calls `prepareNewBuild`, which in turn copies over the `watchFileSystem` value. So it does copy over the new value, as long as you have a `StartParameterInternal`.